app/services/slug_setup.rb

```diff
@@ -0,0 +1,58 @@
+class SlugSetup
+    
+    attr_reader :name 
+
+    def self.setup(name)
+        new(name).send(:candidates)
+    end
+
+    def initialize(name)
+        @name = name
+    end
+    
+    def candidates
+        [   @name.method(:short_name), 
+            @name.method(:prolonged_name), 
+            @name.method(:orged), 
+            :name
+        ]
+    end
+    
+end
+
+class String
+    
+    NOT_WANTED = %w(the of for and in to)
+
+    def short_name
+        if self.match(/parochial church/i)
+            parochial_churches
+        else
+            slug_words.first(3).join('-')
+        end
+    end
+
+    def prolonged_name
+        "#{short_name}-#{slug_words[-2]}-#{slug_words[-1]}"
+    end
+
+    def orged
+        "#{slug_words.join('-')}-org"
+    end
+
+    def slug_words
+        words = "#{self}".delete('\'','-').downcase_words
+        words.reject {|w| NOT_WANTED.include?(w) }
+    end
+    
+    def parochial_churches
+        parish = self.slice(/(?<=parish of )(.+)\z/i).downcase_words
```

Comment:
I assumed `self` is necessary here. But if it's not necessary, I guess it's helpful, because it's more explicit.